src/components/primitive/text/text.component.tsx

```diff
@@ -1,77 +1,77 @@
+    const i18nText = tx && translate(tx, txOptions);
+    const content = i18nText || text || children;
 
-    case TextPresets.SMALL:
-      TextComponent = Small;
-      break;
+    let TextComponent;
 
-    default:
-      TextComponent = Base;
-  }
+    switch (variant) {
+      case TextPresets.TITLE:
+        TextComponent = Title;
+        break;
+      case TextPresets.CONTENT_HEADER:
+        TextComponent = ContentHeader;
+        break;
+      case TextPresets.CONTENT:
+        TextComponent = Content;
+        break;
+
+      case TextPresets.SMALL:
+        TextComponent = Small;
+        break;
 
-  const Typography = styled(TextComponent)`
-    /* custom styles goes here */
+      default:
+        TextComponent = Base;
+    }
 
-    text-transform: ${({ upperCase }) => (upperCase ? `uppercase` : '')};
-    text-transform: ${({ lowerCase }) => (lowerCase ? `lowercase` : '')};
-    text-transform: ${({ capitalize }) => (capitalize ? `lowercase` : '')};
-    text-align: ${({ center }) => (center ? `center` : '')};
-    text-align: ${({ left }) => (left ? `left` : '')};
-    text-align: ${({ right }) => (right ? `right` : '')};
-    line-height: ${({ lineHeight }) => (lineHeight ? `${lineHeight}em` : '')};
-    padding-left: ${({ paddingLeft }) => (paddingLeft ? ` ${paddingLeft}rem` : '')};
-    padding-right: ${({ paddingRight }) => (paddingRight ? ` ${paddingRight}rem` : '')};
-    font-weight: ${({ noBold }) => (noBold ? `normal` : '')};
-    font-size: ${({ size }) => (size ? `${size}rem` : '')};
-  `;
+    const Typography = styled(TextComponent)`
+      /* custom styles goes here */
 
-  return (
-    <Typography
-      upperCase={upperCase}
-      lowerCase={lowerCase}
-      capitalize={capitalize}
-      center={center}
-      left={left}
-      right={right}
-      size={size}
-      paddingLeft={paddingLeft}
-      paddingRight={paddingRight}
-      variant={variant}
-    >
-      {children}
-    </Typography>
-  );
-};
+      text-transform: ${({ upperCase }) => (upperCase ? `uppercase` : '')};
+      text-transform: ${({ lowerCase }) => (lowerCase ? `lowercase` : '')};
+      text-transform: ${({ capitalize }) => (capitalize ? `lowercase` : '')};
+      text-align: ${({ center }) => (center ? `center` : '')};
+      text-align: ${({ left }) => (left ? `left` : '')};
+      text-align: ${({ right }) => (right ? `right` : '')};
+      line-height: ${({ lineHeight }) => (lineHeight ? `${lineHeight}em` : '')};
+      padding-left: ${({ paddingLeft }) => (paddingLeft ? ` ${paddingLeft}rem` : '')};
+      padding-right: ${({ paddingRight }) => (paddingRight ? ` ${paddingRight}rem` : '')};
+      font-weight: ${({ noBold }) => (noBold ? `normal` : '')};
+      font-weight: ${({ bold600 }) => (bold600 ? `600` : '')};
+      font-size: ${({ size }) => (size ? `${FontSizes[size]}rem` : '')};
+      color: ${colors[color]} !important;
+    `;
 
-export { Text };
+    return <Typography {...rest}>{content}</Typography>;
```

Comment:
Rest props are passed only when the styled component doesn't have access to component props. Please move the Typography component out if you are using this syntax.